### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by eugene-eeo**
About the change to **.gitignore**: I needed to install the package locally and so it generated some `.egg-info` directories. Most developers who want to work on the source will already do this anyways so I added it.

### CODE REVIEW

**Comment by saghul** [comment]
Something is weird with Travis. I'll fix that and then you can rebase your patches and have them tested.

**Comment by saghul** [comment]
@eugene-eeo can you please rebase and force push? Travis should be fixed now.

**Comment by eugene-eeo** [comment]
@saghul How do I do that?

**Comment by saghul** [comment]
Tests fail on PyPy: https://travis-ci.org/saghul/python-fibers/jobs/77135838

**Comment by eugene-eeo** [comment]
Yes; I'll take a look at the old code to see if there's some (platform specific) logic I missed out.

**Comment by saghul** [comment]
FWIW, all your changes affect on PyPy, since you modified _pyfibers. CPython uses the C extension.

**Comment by eugene-eeo** [comment]
Is there a way to cherry pick commits and test them against Travis?

**Comment by saghul** [comment]
Just keep adding commits here, we can squash and rebase later on. Thanks!

File: fibers/_pyfibers.py [hunk 1]
```diff
@@ -50,13 +50,25 @@ def _run(c):
             raise error('parent cannot be on a different thread')
         self.parent = parent
 
+    def _get_active_parent(self):
+        parent = self.parent
+        while True:
+            if (parent is not None
+                    and parent._cont is not None
+                    and not parent._ended):
+                break
```
Comment (saghul) [inline fibers/_pyfibers.py]: I think you're missing a `parent = parent.parent` after the break. Then just return parent.

Comment (eugene-eeo) [inline fibers/_pyfibers.py]: Thanks! Will fix it tomorrow. Can you just mention my username so it shows up in my notifications?

Comment (saghul) [inline fibers/_pyfibers.py]: @eugene-eeo no problem, I don't plan to make any changes myself before merging this.

**Comment by eugene-eeo** [comment]
@saghul I'm done with the PR, you can go ahead and merge it now.

**Comment by saghul** [comment]
@eugene-eeo Build is kosher now, thanks! I'll merge it later today!

**Comment by saghul** [comment]
Landed in 5cc65f45c0dba4c47015d1b722ceb5b3ae391c83, c1068e3a97f8af2b26396ce1402a708798b4c007, c1068e3a97f8af2b26396ce1402a708798b4c007 and c1068e3a97f8af2b26396ce1402a708798b4c007. Thanks!

**Pull request closed without merging**